services/QuillLMS/app/services/payload_hasher.rb

```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+class PayloadHasher < ApplicationService
+  attr_reader :payload
+
+  def initialize(payload)
```

Comment:
One small naming nit, `payload` isn't a generic param, since you are doing `.join('-')` it needs to be an `array`, you might want to name it as such.